Common/src/main/java/at/petrak/hexcasting/common/blocks/akashic/BlockEntityAkashicBookshelf.java

```diff
@@ -89,4 +92,20 @@ protected void loadModData(CompoundTag tag) {
             this.iotaTag = null;
         }
     }
+
+    @Override
+    public CompoundTag readIotaTag(){
+        if(pattern == null){
+            return IotaType.serialize(emptyIota());
+        }
+        return IotaType.serialize(new PatternIota(pattern));
+    }
+
+    @Override
+    public Iota readIota(ServerLevel world) {
+        if(pattern == null){
+            return emptyIota();
+        }
+        return new PatternIota(pattern);
+    }
```

Comment:
It feels weird for the logic here to be duplicated... not sure how to fix it though.